src/mimefactory.rs

```diff
@@ -555,9 +553,22 @@ impl<'a> MimeFactory<'a> {
             render_rfc724_mid(&rfc724_mid),
         ));
 
-        headers
-            .unprotected
-            .push(Header::new_with_value("To".into(), to).unwrap());
+        let undisclosed_recipients = match &self.loaded {
+            Loaded::Message { chat } => chat.typ == Chattype::Broadcast,
+            Loaded::Mdn { .. } => false,
+        };
+
+        if undisclosed_recipients {
+            headers.unprotected.push(Header::new(
+                "To".into(),
+                "undisclosed-recipients: ;".to_string(),
```

Comment:
https://github.com/rspamd/rspamd/blob/master/rules/regexp/headers.lua#L109 looks to me as if `undisclosed-recipients: ;` is considered spammy but e.g. `hidden-recipients: ;` (or any other string) is not. Which seems weird, but if true, we could just use the latter.

@whalehub do you know more about this?

Reply:
https://github.com/deltachat/deltachat-core-rust/pull/1939/files#diff-3b6eee8bbfb408198141f58245e321edbf52aa5b01612806f9693e292c3b9ae5R567 did use `hidden_recipients:`(with underscore), which looks weird to me, therefore i checked thunderbird. but we can change that, sure. an option is also to send `To:`==`From:`.

Reply:
Let's wait for @whalehub to comment, maybe he can tell us what is preferable

Reply:
@Hocuri @r10s I'm a bit confused by the ping - I don't know anything about this.

Reply:
As you had opened https://github.com/deltachat/deltachat-core-rust/issues/1575 and provided some links there (which is the only reason I know about https://github.com/rspamd/rspamd/blob/master/rules/regexp/headers.lua), I had the impression you know how to read this format?

The question is, is it true that https://github.com/rspamd/rspamd/blob/master/rules/regexp/headers.lua#L109 means that `undisclosed-recipients: ;` is considered spammy while e.g. `hidden-recipients: ;` (or any other string) is fine?

If you don't know, we can merge it as-is and then you do the same tests as you did for #1575?

Reply:
@Hocuri Oh right, I remember now. As far as I can tell, your interpretation is correct. Using the `undisclosed-recipients: ;` header increases the spam score of an e-mail in Rspamd by 3, which is significant. Using a different string like `hidden-recipients: ;` instead should be fine, as long as the `To:` field isn't left blank. I'll be happy to test it once this pull request is merged and included in a new release.

Reply:
Cool, thanks!